pkg/reconciler/autoscaling/kpa/scaler.go

```diff
@@ -198,8 +199,12 @@ func (ks *scaler) handleScaleToZero(ctx context.Context, pa *autoscalingv1alpha1
 	switch {
 	case pa.Status.IsActivating(): // Active=Unknown
 		// If we are stuck activating for longer than our progress deadline, presume we cannot succeed and scale to 0.
-		if pa.Status.CanFailActivation(now, activationTimeout) {
-			logger.Info("Activation has timed out after ", activationTimeout)
+		// Alternatively, if there is no traffic routed to the revision, and we are stuck activating for longer than the
+		// activationTimeoutBuffer, then scale to 0
+		if pa.Status.CanFailActivation(now, activationTimeout) ||
+			(!routesToRevision && pa.Status.CanFailActivation(now, activationTimeoutBuffer)) {
```

Comment:
`activationTimeout` already includes the `activationTimeoutBuffer` see lines. 190-195

Reply:
Yes. But, given there is no routes to the revision. I still need to wait some time before deciding to scale to 0. Currently this waiting time is the `progressDeadline (600secs) +  activationTimeoutBuffer (30secs)`, which is too much, given there is no routes to the revision.
But, if there is no routes to the revision and I scale to 0 right away, I end up scaling to 0 all revisions (even the "good ones". We do not want that.
I thought: let's wait some time (activationTimeoutBuffer) and if there is no routes, then scale to 0.

Reply:
I see - then for readability can we make this as `else if` clause - then we can have distinct log messages for the different cases

Reply:
done